borg/constants.py

```diff
@@ -12,8 +12,14 @@
 CACHE_TAG_NAME = 'CACHEDIR.TAG'
 CACHE_TAG_CONTENTS = b'Signature: 8a477f597d28d172789f06886806bc55'
 
-DEFAULT_MAX_SEGMENT_SIZE = 5 * 1024 * 1024
-DEFAULT_SEGMENTS_PER_DIR = 10000
+# A large, but not unreasonably large segment size. Always less than 2 GiB (for legacy file systems). We choose
+# 500 MiB which means that no indirection from the inode is needed for typical Linux file systems.
+# Note that this is a soft-limit and can be exceeded (worst case) by a full maximum chunk size and some metadata
+# bytes. That's why it's 500 MiB instead of 512 MiB.
+DEFAULT_MAX_SEGMENT_SIZE = 500 * 1024 * 1024
```

Comment:
not that I have a strong opinion, but since segments are still made out of chunks, you do not need to keep the whole segment file in RAM, what's to stop you from reading it chunk by chunk, do the conversion and write it out into the other file?

500MB file is not all that big and looks reasonably sized for me, works on all filesystem types too.

On the other hand 50MB files is probably ok too, you'll just have more of them => a bit more of directory overhead, 10x more inodes used and such, but probably nothing fatal.